pkg/deploy/controller/generictrigger/controller.go

```diff
@@ -36,183 +29,19 @@ type DeploymentTriggerController struct {
 	codec runtime.Codec
 }
 
-// fatalError is an error which can't be retried.
-type fatalError string
-
-func (e fatalError) Error() string {
-	return fmt.Sprintf("fatal error handling configuration: %s", string(e))
-}
-
 // Handle processes deployment triggers for a deployment config.
 func (c *DeploymentTriggerController) Handle(config *deployapi.DeploymentConfig) error {
```

Comment:
Can you reshape this into the `NewFooController()`, `Run(workers, stopCh)`, `EnqueueDC()` shape?  Splitting across files is driving me bonkers.